modules/oe_content_event/tests/src/Kernel/EventLinkSourceFilterTest.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_content_event\Kernel;
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+use Drupal\node\Entity\Node;
+use Drupal\oe_content_event\Plugin\InternalLinkSourceFilter\EventPeriodFilter;
+
+/**
+ * Tests the internal link source filters related to Events.
+ *
+ * @covers \Drupal\oe_content_event\Plugin\InternalLinkSourceFilter\EventPeriodFilter
+ */
+class EventLinkSourceFilterTest extends EventKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'oe_link_lists',
+    'oe_link_lists_internal_source',
+    'datetime_testing',
+    'oe_time_caching',
+  ];
+
+  /**
+   * Test that the event filter works as intended.
+   */
+  public function testEventLinkSourceFilter(): void {
+    // Freeze the time at a specific point.
+    $static_time = new DrupalDateTime('2020-02-17 14:00:00', DateTimeItemInterface::STORAGE_TIMEZONE);
+    /** @var \Drupal\Component\Datetime\TimeInterface $datetime */
+    $time = $this->container->get('datetime.time');
+    $time->freezeTime();
+    $time->setTime($static_time->getTimestamp());
```

Comment:
We can set time one right before we run time related assertions, so the test becomes more readable.